test/e2e/helpers.go

```diff
@@ -552,6 +553,123 @@ func WaitForControlPlaneAndMachinesReady(ctx context.Context, input WaitForContr
 	})
 }
 
+type ApplyInPlaceUpgradeAndWaitInput struct {
+	Getter                  framework.Getter
+	Machine                 *clusterv1.Machine
+	ClusterProxy            framework.ClusterProxy
+	WaitForUpgradeIntervals []interface{}
+}
+
+func ApplyInPlaceUpgradeAndWait(ctx context.Context, input ApplyInPlaceUpgradeAndWaitInput) {
+	mgmtClient := input.ClusterProxy.GetClient()
+
+	patchHelper, err := patch.NewHelper(input.Machine, mgmtClient)
+	Expect(err).ToNot(HaveOccurred())
+	mAnnotations := input.Machine.GetAnnotations()
+
+	if mAnnotations == nil {
+		mAnnotations = map[string]string{}
+	}
+
+	mAnnotations[bootstrapv1.InPlaceUpgradeToAnnotation] = "localPath=/k8s/upgrade/bin/kubernetes"
+	input.Machine.SetAnnotations(mAnnotations)
+	err = patchHelper.Patch(ctx, input.Machine)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Checking for in-place upgrade status to be equal to done")
+
+	Eventually(func() (bool, error) {
+		um := &clusterv1.Machine{}
+		if err := input.Getter.Get(ctx, client.ObjectKey{Namespace: input.Machine.Namespace, Name: input.Machine.Name}, um); err != nil {
+			Byf("Failed to get the machine: %+v", err)
+			return false, err
+		}
+
+		mAnnotations := um.GetAnnotations()
+
+		status, ok := mAnnotations[bootstrapv1.InPlaceUpgradeStatusAnnotation]
+		if !ok {
+			return false, nil
+		}
+
+		return status == bootstrapv1.InPlaceUpgradeDoneStatus, nil
+	}, input.WaitForUpgradeIntervals...).Should(BeTrue(), "In-place upgrade failed for %s", input.Machine.Name)
+}
+
+type ApplyInPlaceUpgradeForControlPlaneInput struct {
+	Lister                  framework.Lister
+	Getter                  framework.Getter
+	ClusterProxy            framework.ClusterProxy
+	Cluster                 *clusterv1.Cluster
+	WaitForUpgradeIntervals []interface{}
+}
+
+func ApplyInPlaceUpgradeForControlPlane(ctx context.Context, input ApplyInPlaceUpgradeForControlPlaneInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for ApplyInPlaceUpgrade")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling ApplyInPlaceUpgradeForControlPlane")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling ApplyInPlaceUpgradeForControlPlane")
```

Comment:
I don't think the extra strings in ToNot() are very useful, they just repeat what we're testing for. For example, we test for `ToNot(BeNil)`, if the condition is not met, we log "It can't be Nil." The dev whose test is failing can simply look up the line that's failing and they'll see `NotTo(BeNil)`, so the text is redundant. I'm in favor of removing these so there's less visual noise.